api/src/main/java/com/velocitypowered/api/event/AsyncTask.java

```diff
@@ -0,0 +1,78 @@
+package com.velocitypowered.api.event;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+@FunctionalInterface
+public interface AsyncTask {
```

Comment:
I'm not a huge fan of the name `AsyncTask`. I don't have another good name for this - perhaps `EventTask` or even `EventContinuation` would be better (and then rename `AsyncTask.WithContinuation` to `EventContinuation.Deferred`).

Reply:
I'm also not a big fan of the current naming, but I haven't found a good name to actually represent what is does. `DeferredEventTask` (`DeferredTask`) could be another option?

Reply:
It needs to include the word `Event` as otherwise it could be confused with the scheduler. I think `DeferredEventTask` is fine.